compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -537,7 +600,7 @@ fn inner_optimized_mir(tcx: TyCtxt<'_>, did: LocalDefId) -> Body<'_> {
 
     debug_assert!(!body.has_free_regions(), "Free regions in optimized MIR");
 
-    body
+    remap_mir_for_const_eval_select(tcx, body, hir::Constness::NotConst)
```

Comment:
Could this run before the optimizations? That would allow inlining the non-const callee, making `const_eval_select` pretty much zero-cost even if the backend doesn't support inlining.

Reply:
> Could this run before the optimizations?

Maybe, but it could create unnecessary memory in query system storage as we now have to pass a `bool` to indicate whether we want the const version or not, right?

Reply:
You can just do it at the beginning of `inner_optimized_mir` rather than at the end, can't you?

Reply:
What @RalfJung suggested is exactly what I meant.